modules/ding_loan/plugins/content_types/loans.inc

```diff
@@ -254,13 +254,47 @@ function ding_loan_loans_form($form, &$form_state, $account, $loans = array()) {
         );
       }
 
-      // If not renewable set message.
+      // This is the same array as in loan form submit further down, but we
+      // might want different feedback messages on the list.
+      $error_messages = array(
+        DingProviderLoan::STATUS_NOT_RENEWED => t('This material can not be renewed'),
+        DingProviderLoan::STATUS_RENEWAL_NUM_EXCEEDED => t('The material can not be renewed because maximum number of renewals reached'),
+        DingProviderLoan::STATUS_RENEWAL_RESERVED => t('The material is reserved by another loaner and can not be renewed'),
+      );
+
+      // Loan is not renewable! Inform the user the best we can.
       if (empty($loan->renewable)) {
         $disabled_count++;
-        $item['#material_message'] = array(
-          'message' => t('This material can not be renewed'),
-          'class' => 'messages warning',
-        );
+        // If already renewed this session inform the user to avoid confusion.
+        if ($loan->renewalStatus == DingProviderLoan::STATUS_RENEWED) {
+          $item['#material_message'] = array(
+            'message' => t('This material was renewed. New return date: !due_date', array(
+              '!due_date' => format_date(strtotime(check_plain($loan->expiry)), 'ding_material_lists_date'),
+            )),
+            'class' => 'messages information',
+          );
+        }
+        // Inter library loan.
+        elseif ($loan->renewalStatus == DingProviderLoan::STATUS_RENEWAL_REQUESTED) {
+          $item['#material_message'] = array(
+            'message' => t('An inter-library loan renewal was requested'),
+            'class' => 'messages information',
+          );
+        }
+        // This is a special case. If the material was borrowed today and is
+        // not renewed/rewenable; we show this message instead.
+        elseif (date('Y-m-d') == date('Y-m-d', strtotime($loan->loan_date))) {
+          $item['#material_message'] = array(
+            'message' => t('This material was borrowed today'),
+            'class' => 'messages information',
+          );
+        }
+        else {
+          $item['#material_message'] = array(
+            'message' => $error_messages[$loan->renewalStatus],
+            'class' => 'messages warning',
+          );
```

Comment:
I am not too fond of this approach using `$error_messages` as an array.

1. The array is defined pretty far away from where it is used.
2. This checks for `$loan->renewalStatus` but there are already two checks on `$loan->renewalStatus`. I guess they cannot be included here because they use a different class?
3. There is no default message

I wouuld prefer if `$error_messages` and replaced with 2 more checks á la `elseif ($loan->renewalStatus == DingProviderLoan::STATUS_RENEWAL_NUM_EXCEEDED)` with a final default `else` using `This material can not be renewed`.

Reply:
I'll remove array and add checks and default.

I'm not sure about question in 2. Will look into it.

Reply:
On further inspection: I guess I was trying to separate the error_messages from the status messages. But I can see it's a bit confusing, so I'll just remove the array and add checks instead.